ui/app/config.js

```diff
@@ -99,6 +110,19 @@ ConfigScreen.prototype.render = function () {
   )
 }
 
+function rpcValidation (newRpc, state) {
+  if (validUrl.isWebUri(newRpc)) {
+    state.dispatch(actions.setRpcTarget(newRpc))
+  } else {
+    var appendedRpc = `http://${newRpc}`
+    if (validUrl.isWebUri(appendedRpc)) {
+      state.dispatch(actions.displayWarning('URIs require the appropriate HTTP/HTTPS prefix.'))
```

Comment:
My reasoning for this is that if the URI validator works the first time, then we're cool. If the URI validator works when we append a simple http:// prefix, then we warn users to use it (we don't automatically append it because we are unsure of whether they want to use an HTTP or HTTPS protocol). If the URI doesn't even work with that, then we just throw a general warning.

So unless we want to just do a simple URI validation, this is a fairly clean way to check if they're just missing an http:// without making too many assumptions.